### DISCUSSION TITLE
List comprehension in C++

### DISCUSSION THREAD

**Discussion by ritam9495** [Z3Prover/z3#5066] [Q&A] [answered]
Using list comprehension in python is easy for a piece of code resembling the one below
`a = [1, 2, 3, 4, 5]`
f(x) is an uninterpreted function
`s.add(And(x >=0, x < len(a)))`
`s.add(ForAll(x, Or([f(x) == a[i] for i in range(0, len(a))])))`

How can something like this be implemented in C++?

**Comment by NikolajBjorner** [comment, marked as answer]
Most typically you use a for-loop and push_back on the expr_vector object. With lambdas you can approach code that looks like comprehensions, but of course, C++ does not support comprehensions.

**Reply by ritam9495** [reply]
thanks. that's helpfull
